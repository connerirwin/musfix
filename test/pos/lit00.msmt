; Uninterpreted Sorts
(declare-sort List 1)
(declare-sort Peano 0) ; manual

; Constants
(declare-const one Peano)
(declare-const zero Peano)


; Distinct Constants
(assert (distinct one zero))


; Uninterpreted Functions


; Qualifiers


; Well-formedness Constraints


; Horn Constraints
(constraint (forall ((xZero Peano)(xOne Peano)(v1 Int)) (=> (and (= xZero zero) (and (= xOne one) True)) (not (= xZero xOne)))))
