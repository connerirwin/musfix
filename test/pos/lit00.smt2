; Uninterpreted Sorts
(declare-sort List 1)
(declare-sort Peano 0)


; Constants
(declare-const one Peano)
(declare-const zero Peano)


; Distinct Constants
(assert (distinct one zero))


; Uninterpreted Functions


; Qualifiers


; Well-formedness constraints


; Horn constraints
(constraint (forall ((xZero Peano)(xOne Peano)(v1 Int)) (=> (and (= xZero zero) (and (= xOne one) True)) (not (= xZero xOne)))))
