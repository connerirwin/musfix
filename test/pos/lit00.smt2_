; Uninterpreted Sorts


; Uninterpreted Functions
(declare-fun One Peano)
(declare-fun Zero Peano)


; Qualifiers


; Well-formedness constraints


; Horn constraints
(constraint (forall ((xZero Peano)(xOne Peano)(v1 Int)) (=> (and (= xZero Zero) (and (= xOne One) True)) (not (= xZero xOne)))))
