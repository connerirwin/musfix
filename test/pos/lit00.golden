
Preparing to find greatest fixpoint...

Inputs
--------
Reading from file test/pos/lit00.smt2
SortDecl "List" 1
SortDecl "Peano" 0
ConstantDecl "one" (DataS "Peano" [])
ConstantDecl "zero" (DataS "Peano" [])
DistinctDecl ["one","zero"]
HornConstraint [Var (DataS "Peano" []) "xZero",Var (DataS "Peano" []) "xOne",Var IntS "v1"] (Binary Implies (Binary And (Binary Eq (Var (DataS "Peano" []) "xZero") (Func (DataS "Peano" []) "zero" [])) (Binary And (Binary Eq (Var (DataS "Peano" []) "xOne") (Func (DataS "Peano" []) "one" [])) (BoolLit True))) (Unary Not (Binary Eq (Var (DataS "Peano" []) "xZero") (Var (DataS "Peano" []) "xOne"))))

QMAP
--------
fromList []

Candidates
--------
[Candidate {solution = fromList [], validConstraints = fromList [Binary Implies (Binary And (Binary Eq (Var (DataS "Peano" []) "xZero") (Func (DataS "Peano" []) "zero" [])) (Binary And (Binary Eq (Var (DataS "Peano" []) "xOne") (Func (DataS "Peano" []) "one" [])) (BoolLit True))) (Unary Not (Binary Eq (Var (DataS "Peano" []) "xZero") (Var (DataS "Peano" []) "xOne")))], invalidConstraints = fromList [], label = "0"}]


Final candidates: 
0: [] (1 0)

